src/core/annotation.js

```diff
@@ -746,6 +739,37 @@ var LinkAnnotation = (function LinkAnnotationClosure() {
   return LinkAnnotation;
 })();
 
+var PopupAnnotation = (function PopupAnnotationClosure() {
+  function PopupAnnotation(parameters) {
+    Annotation.call(this, parameters);
+
+    this.data.annotationType = AnnotationType.POPUP;
+
+    var dict = parameters.dict;
+    if (!dict.has('Parent')) {
+      warn('Popup annotations without a parent annotation are not supported.');
+      return;
+    }
+
+    var parentItem = dict.get('Parent');
+    if (!parentItem) {
+      warn('Popup annotation has an invalid parent annotation.');
+      return;
+    }
+    this.data.parentId = dict.map.Parent.num;
```

Comment:
You probably want [`Dict_getRaw`](https://github.com/mozilla/pdf.js/blob/master/src/core/primitives.js#L160-L163), since it looks like you want to access a [`Reference`](https://github.com/mozilla/pdf.js/blob/master/src/core/primitives.js#L259-L275). And I don't think that you should use `objId` when you have access to a reference, since it's also somewhat private, and should probably not be relied on.

But the real question here is: if you have a `Ref`, why do you directly access just *one* of its properties? In practice, the `gen` is usually zero, but you cannot assume that that *always* holds, so I really think that you ought to use `Ref_toString` instead!

This leads me to look at the current code in `annotation.js`, and I thus think that https://github.com/mozilla/pdf.js/blob/master/src/core/annotation.js#L163 is actually wrong! That line really ought to read `this.data.id = params.ref.toString();`, which means that changing the current line to `this.data.parentId = dict.getRaw('Parent').toString()` should then work just fine.